File: plugins/rd/skills/10-stages-developing/templates/typescript/test-template.ts
/**
 * Tests for module-name
 */

import { functionName } from './module-name';

describe('functionName', () => {
    test('smoke test', () => {
        const result = functionName('test', 10);
        expect(result).toBeDefined();
        expect(result).toHaveProperty('key1');
        expect(result).toHaveProperty('key2');
    });

    test('processes valid input correctly', () => {
        const result = functionName('hello', 42);

        expect(result.key1).toBe('processed_hello');
        expect(result.key2).toBe(84);
    });

    test('throws error for empty string', () => {
        expect(() => functionName('', 10))
            .toThrow('param1 cannot be empty');
    });

    test('throws error for negative number', () => {
        expect(() => functionName('test', -1))
            .toThrow('param2 must be non-negative');
    });

    test('handles zero correctly', () => {
        const result = functionName('test', 0);
        expect(result.key2).toBe(0);
    });

    test('handles large numbers', () => {
        const result = functionName('test', 1000000);
        expect(result.key2).toBe(2000000);
    });

    // Parametrized tests
    test.each([
        ['hello', 10, 'processed_hello', 20],
        ['world', 5, 'processed_world', 10],
        ['test', 0, 'processed_test', 0],
    ])('processes %s and %d correctly', (inputStr, inputNum, expectedKey1, expectedKey2) => {
        const result = functionName(inputStr, inputNum);
        expect(result.key1).toBe(expectedKey1);
        expect(result.key2).toBe(expectedKey2);
    });
});

File: skills/10-stages-developing/templates/python/test-template.py
"""
Tests for [module_name].

This module contains comprehensive tests for [module_name] functions.
"""

import pytest
from module_name import function_name


class TestFunctionName:
    """Test suite for function_name."""

    def test_function_name_smoke(self):
        """Smoke test for function_name."""
        result = function_name("test", 10)
        assert result is not None
        assert isinstance(result, dict)

    def test_function_name_valid_input(self):
        """Test with valid input parameters."""
        result = function_name("hello", 42)

        assert result["key1"] == "processed_hello"
        assert result["key2"] == 84

    def test_function_name_empty_string(self):
        """Test with empty string parameter."""
        with pytest.raises(ValueError, match="cannot be empty"):
            function_name("", 10)

    def test_function_name_negative_number(self):
        """Test with negative number parameter."""
        with pytest.raises(ValueError, match="must be non-negative"):
            function_name("test", -1)

    def test_function_name_zero(self):
        """Test with zero as parameter."""
        result = function_name("test", 0)
        assert result["key2"] == 0

    def test_function_name_large_number(self):
        """Test with large number parameter."""
        result = function_name("test", 1000000)
        assert result["key2"] == 2000000


# Parametrized tests for multiple inputs
@pytest.mark.parametrize("input_str,input_num,expected_key1,expected_key2", [
    ("hello", 10, "processed_hello", 20),
    ("world", 5, "processed_world", 10),
    ("test", 0, "processed_test", 0),
])
def test_function_name_parametrized(input_str, input_num, expected_key1, expected_key2):
    """Test with multiple parameter combinations."""
    result = function_name(input_str, input_num)
    assert result["key1"] == expected_key1
    assert result["key2"] == expected_key2
